### ISSUE TITLE
Add embedding lookup to GraphSAGE and HinSAGE

### ISSUE THREAD

**Issue by adocherty** [stellargraph/stellargraph#69]
### Description
To train the GraphSAGE or HinSAGE model without node attributes we currently need use a one-hot input feature matrix which is computationally intractable for a large number of nodes. We need to use an embedding layer (https://keras.io/layers/embeddings/)

### Done Checklist (Development)
- [ ] Produced code for required functionality
- [ ] Branch and Pull Request build on CI
- [ ] Branch and Pull Request pass unit tests on CI
- [ ] Branch and Pull Request pass integration tests on CI
- [ ] Version number reflects new status
- [ ] Peer Code Review Performed
- [ ] Code well commented

**Comment by youph** [comment]
@geoffj-d61 would you be willing to take this on?

**Comment by geoffj-d61** [comment]
@youph , is this to be a general change for all models, or just a special case for models without features? I haven't looked much at the code yet, but I do know Keras requires any embedding layer to be the first layer.

**Comment by geoffj-d61** [comment]
@youph @adocherty, is there a design document anywhere to help me understand the GraphSAGE implementation? I can read the code, but can't seem to get a clear picture of how or why it hangs together.

**Comment by geoffj-d61** [comment]
I'm concerned about the use-case for this issue. If we take a graph without features and add arbitrary features via an embedding (or even via the identity matrix approach), then surely we now have a transductive approach that won't be applicable to a new graph?

**Comment by youph** [comment]
> I'm concerned about the use-case for this issue. If we take a graph without features and add arbitrary features via an embedding (or even via the identity matrix approach), then surely we now have a transductive approach that won't be applicable to a new graph?

True, it would then be inductive. The use case would be something like: unsupervised representation learning on a graph with featureless nodes - same use case as DeepWalk or node2vec. You're probably right that no-one should use GraphSAGE like this, they should rather use DeepWalk or node2vec, but the reason this ticket was created is to cover all possible use cases :)
